package/tests/test_api/test_graphql.py

```diff
@@ -1,5 +1,186 @@
+                "col6": [{"runId": "2021-11-02T18.24.24.379Z", "value": 6.0}],
+            },
+        ),
+        TrackingDataSet(
+            datasetName="json_tracking",
+            datasetType="kedro.extras.datasets.tracking.json_dataset.JSONDataSet",
+            data={
+                "col2": [{"runId": "2021-11-02T18.24.24.379Z", "value": True}],
+                "col3": [{"runId": "2021-11-02T18.24.24.379Z", "value": 3}],
+                "col7": [
+                    {"runId": "2021-11-02T18.24.24.379Z", "value": "column_seven"}
+                ],
+            },
+        ),
+    ]
+
+
+class TestTrackingData:
+    def test_graphql_run_tracking_data_query(
+        self, tmp_path, data_access_manager: DataAccessManager, example_tracking_output
+    ):
+        save_version = "2021-11-02T18.24.24.379Z"
+        with mock.patch(
+            "kedro_viz.api.graphql.data_access_manager", new=data_access_manager
+        ):
+            metrics_dataset = MetricsDataSet(
+                filepath=Path(tmp_path / "test.json").as_posix(),
+                version=Version(None, save_version),
+            )
+            metrics_dataset.save({"col1": 1, "col2": 2, "col3": 3})
+
+            dataset = CSVDataSet(filepath="dataset.csv")
+
+            more_metrics = MetricsDataSet(
+                filepath=Path(tmp_path / "metrics.json").as_posix(),
+                version=Version(None, save_version),
+            )
+            more_metrics.save({"col4": 4, "col5": 5, "col6": 6})
+
+            json_dataset = JSONDataSet(
+                filepath=Path(tmp_path/ "tracking.json").as_posix(),
+                version=Version(None, save_version),
+            )
+            json_dataset.save({"col7": "column_seven", "col2": True, "col3": 3})
+
+            catalog = DataCatalog(
+                data_sets={
+                    "metrics": metrics_dataset,
+                    "csv": dataset,
+                    "more_metrics": more_metrics,
+                    "json_tracking": json_dataset,
+                }
+            )
+            data_access_manager.add_catalog(catalog)
+
+            assert get_run_tracking_data([ID(save_version)]) == example_tracking_output
+
+    @patch("logging.Logger.warning")
+    def test_graphql_run_no_tracking_data_query(
+        self,
+        patched_warning,
+        data_access_manager: DataAccessManager,
+    ):
+        save_version = "2021-11-02T18.24.24.379Z"
+        with mock.patch(
+            "kedro_viz.api.graphql.data_access_manager", new=data_access_manager
+        ):
+            dataset = CSVDataSet(filepath="dataset.csv")
+            catalog = DataCatalog(
+                data_sets={
+                    "csv": dataset,
+                }
+            )
+            data_access_manager.add_catalog(catalog)
+
+            assert get_run_tracking_data([ID(save_version)]) == []
+
+            patched_warning.assert_has_calls(
```

Comment:
I think this warning might not be necessary. Once we have the design, this is a legit path that we need to return some response to the client. I'd ignore it for now.